examples/data/mysql/src/main/java/io/helidon/examples/data/mysql/PokemonService.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    /**
+     * Handles the {@code GET /all} request and returns a list of all {@link Pokemon} entities.
+     *
+     * @param request  the server request
+     * @param response the server response
+     */
+    private void all(ServerRequest request, ServerResponse response) {
+        response.send(pokemonRepository.listOrderByName());
+    }
+
+    /**
+     * Handles the {@code GET /type/{name}} request and returns a list of {@link Pokemon} entities
+     * associated with a specific breed name.
+     *
+     * @param request  the server request
+     * @param response the server response
+     */
+    private void type(ServerRequest request, ServerResponse response) {
+        String name = request.path().pathParameters().get("name");
+        response.send(pokemonRepository.listByType_Name(name));
+    }
+
+    /**
+     * Handles the {@code GET /get/{name}} request and returns a {@link Pokemon} entity by its name.
+     *
+     * @param request  the server request
+     * @param response the server response
+     */
+    private void pokemon(ServerRequest request, ServerResponse response) {
+        String name = request.path().pathParameters().get("name");
+        response.send(pokemonRepository.findByName(name));
+    }
+
+    /**
+     * Handles the {@code POST /} request and inserts a new {@link Pokemon} entity.
+     * <p>
+     * Pokémon entity content is supplied as JSON object and translated to {@link PokemonDto} class by HTTP
+     * request processing layer.
+     * Pokémon type record must already exist. Pokémon entity must not exist in the database.
+     *
+     * @param pokemonDto the pokémon to insert into the database
+     * @param response   the server response
+     */
+    private void insert(PokemonDto pokemonDto, ServerResponse response) {
+        // Pokemon data modification task is executed as single database transaction
+        response.send(Tx.transaction(() -> {
+            // Retrieve Type entity for provided name
+            Type type = typeRepository.getByName(pokemonDto.type());
+            // Prepare and insert new Pokemon entity into the database
+            Pokemon pokemonEntity = new Pokemon(pokemonDto.name(), type);
+            return pokemonRepository.insert(pokemonEntity);
+        }));
+    }
+
+    /**
+     * Handles the {@code DELETE /{id}} request and deletes a {@link Pokemon} entity by its ID.
+     *
+     * @param request  the server request
+     * @param response the server response
+     */
+    private void delete(ServerRequest request, ServerResponse response) {
+        int id = request.path()
+                .pathParameters()
+                .first("id").map(Integer::parseInt)
+                .orElseThrow(() -> new BadRequestException("No pokémon id"));
+        // Rely on internal transaction handling for this simple data modification task
+        response.send(new DmlResult(pokemonRepository.deleteById(id), "DELETE"));
```

Comment:
This endpoint does not need to return JSON, it can return a simple string instead.

Reply:
I thought that REST service output should be consistent, e.g. return at least the same media type. But as you wish.